src/sql/src/parsers/create_parser.rs

```diff
@@ -246,6 +243,101 @@ impl<'a> ParserContext<'a> {
         Ok((columns, constraints))
     }
 
+    fn parse_column(
+        &mut self,
+        columns: &mut Vec<ColumnDef>,
+        constraints: &mut Vec<TableConstraint>,
+    ) -> Result<()> {
+        let column = self
+            .parser
+            .parse_column_def()
+            .context(SyntaxSnafu { sql: self.sql })?;
+
+        if !matches!(column.data_type, DataType::Timestamp)
+            || matches!(self.parser.peek_token(), Token::Comma)
+        {
+            columns.push(column);
+            return Ok(());
+        }
+
+        // for supporting `ts TIMESTAMP TIME INDEX,` syntax.
+        self.parse_time_index(column, columns, constraints)
+    }
+
+    fn parse_time_index(
+        &mut self,
+        mut column: ColumnDef,
+        columns: &mut Vec<ColumnDef>,
+        constraints: &mut Vec<TableConstraint>,
+    ) -> Result<()> {
+        self.parser
+            .expect_keyword(Keyword::TIME)
+            .context(error::UnexpectedSnafu {
+                sql: self.sql,
+                expected: "TIME",
+                actual: self.peek_token_as_string(),
+            })?;
```

Comment:
Could we use [`expect_keywords()`](https://docs.rs/sqlparser/latest/sqlparser/parser/struct.Parser.html#method.expect_keywords)?
```suggestion
        self.parser
            .expect_keyword(&[Keyword::TIME, Keyword::INDEX])
            .context(error::UnexpectedSnafu {
                sql: self.sql,
                expected: "TIME INDEX",
                actual: self.peek_token_as_string(),
            })?;
```